tenant/store.go

```diff
@@ -49,3 +48,149 @@ type Segment struct {
 	Name     string
 	Seq      uint64
 }
+
+func (tenantStore *tenantStore) listTenants() ([]Tenant, error) {
+	var tenants []Tenant
+	log.Println("In listTenants()", &tenants)
+	tenantStore.DbStore.Db.Find(&tenants)
+	err := common.MakeMultiError(tenantStore.DbStore.Db.GetErrors())
+	if err != nil {
+		return nil, err
+	}
+	log.Println(tenants)
+	return tenants, nil
+}
+
+func (tenantStore *tenantStore) listSegments(tenantId uint64) ([]Segment, error) {
+	var segments []Segment
+	log.Println("In listTenants()", &segments)
+	tenantStore.DbStore.Db.Where("tenant_id = ?", tenantId).Find(&segments)
+	err := common.MakeMultiError(tenantStore.DbStore.Db.GetErrors())
+	if err != nil {
+		return nil, err
+	}
+	log.Println(segments)
+	return segments, nil
+}
+
+func (tenantStore *tenantStore) addTenant(tenant *Tenant) error {
+	var tenants []Tenant
+	tenantStore.DbStore.Db.Find(&tenants)
+
+	//	myId := tenant.Id
+	tenant.Seq = uint64(len(tenants) + 1)
```

Comment:
Since our available numbers are somewhat limited (these get translated into the bits in the IP address, right?) we are forced to recycle the numbers, though.

Segments are a rare resource and especially once we move to a more dynamic segment pool, we will have to consider allowing for deletion (or at least occasional garbage collection).

If we don't want to tackle this in the last few remaining days, which I understand, we should (a) leave a TODO note here, and (b) definitely create a task card to remind us of this.

Also put a comment at the same place where this is done for tenants. They are less dynamic, but at least put a comment in there about what's happening.